pkg/quarterdeck/db/models/apikeys.go

```diff
@@ -376,10 +396,34 @@ func (k *APIKey) Create(ctx context.Context) (err error) {
 	for _, permission := range k.permissions {
 		permparams[1] = sql.Named("permission", permission)
 		if _, err = tx.Exec(insertKeyPermSQL, permparams...); err != nil {
+			var dberr sqlite3.Error
+			if errors.As(err, &dberr) {
+				if dberr.Code == sqlite3.ErrConstraint {
+					return invalid(ErrInvalidPermission)
+				}
+			}
 			return err
 		}
 	}
 
+	// Update the query with its partial status
+	// Set the partial flag by query on the struct
+	if _, err = tx.Exec(updatePartialSQL, sql.Named("keyID", k.ID)); err != nil {
+		var dberr sqlite3.Error
+		if errors.As(err, &dberr) {
+			if dberr.Code == sqlite3.ErrConstraint {
+				return constraint(dberr)
+			}
+		}
+		return err
+	}
+
+	// Fetch the partial status from database
+	// NOTE: this is done this way to allow for logicless transactions in Raft replicated SQLite queries
+	if err = tx.QueryRow(queryPartialSQL, sql.Named("keyID", k.ID)).Scan(&k.Partial); err != nil {
+		return err
+	}
```

Comment:
This will allow us to set the partial in a SQL statement which is required for replicated sqlite with Raft. The second query ensures that Partial is set on the struct.